### PR TITLE
cluster: skip sig verification for versions earlier than v1.3.0

### PR DESCRIPTION
**Pull request by xenowits**
* For older definition versions (`< v1.3.0`):
* Skip verification of operator enr and config signatures 
* Error with "older version signatures not supported" when unmarshalling old definitions
* Add tests

category: feature
ticket: https://github.com/ObolNetwork/charon/issues/1203

### CODE REVIEW

File: cluster/definition.go [hunk 1]
```diff
@@ -139,6 +140,12 @@ func (d Definition) NodeIdx(pID peer.ID) (NodeIdx, error) {
 
 // VerifySignatures returns true if all config signatures are fully populated and valid. A verified definition is ready for use in DKG.
 func (d Definition) VerifySignatures() error {
+	// Skip signature verification for definition versions earlier than v1.3. This is because v1.3 introduced a breaking change to how EIP712
+	// signatures are calculated. https://github.com/ObolNetwork/charon/issues/1203
+	if semver.Compare(d.Version, v1_3) == -1 {
+		return nil
+	}
```
Comment (corverroos) [inline cluster/definition.go:147]: best not introduce additional libraries without consulting with the team, please remove and rather use simple comparison: `if d.Version==v1_0 || d.Version==v1_1 || d.Version==v1_2`.

Comment (corverroos) [reply cluster/definition.go:147]: Also, as mentioned earlier please error if signatures are present in this case, i.e., require empty signatures

Comment (corverroos) [reply cluster/definition.go:147]: can extract the above condition into a function `supportEIP712Sigs`

File: cluster/cluster_test.go [hunk 2]
```diff
@@ -49,20 +50,29 @@ func TestEncode(t *testing.T) {
 					{
 						Address:         testutil.RandomETHAddress(),
 						ENR:             fmt.Sprintf("enr://%x", testutil.RandomBytes32()),
-						ConfigSignature: testutil.RandomSecp256k1Signature(),
-						ENRSignature:    testutil.RandomSecp256k1Signature(),
+						ConfigSignature: []byte{},
```
Comment (corverroos) [inline cluster/cluster_test.go:53]: Suggest adding them as that is default, then removing if version doesn't support it.

**Comment by dB2510** [comment]
Cleared and completed in #1226

**Pull request closed without merging**